ksql-engine/src/main/java/io/confluent/ksql/QueryEngine.java

```diff
@@ -68,23 +60,30 @@
   // CHECKSTYLE_RULES.ON: ClassDataAbstractionCoupling
 
   private static final Logger log = LoggerFactory.getLogger(QueryEngine.class);
-  private final KsqlEngine ksqlEngine;
-  private final DdlCommandFactory ddlCommandFactory;
-
-
-  QueryEngine(final KsqlEngine ksqlEngine, final DdlCommandFactory ddlCommandFactory) {
-    this.ddlCommandFactory = ddlCommandFactory;
-    this.ksqlEngine = ksqlEngine;
+  private final KafkaTopicClient topicClient;
+  private final Supplier<SchemaRegistryClient> schemaRegistryClientFactory;
+  private final QueryIdGenerator queryIdGenerator;
+  private final QueryIdGenerator tryQueryIdGenerator;
```

Comment:
Due to the validation following the same code path as the execution, we need to have separate instances of `QueryIdGenerator` for validation vs execution. Otherwise, the id's of queries will change with this PR.  To stop clashes in the query ids the two generate I've added a postfix.